lib/repo/repository.go

```diff
@@ -278,115 +85,59 @@ func (g *GenericRepository) Name() string {
 	return tokens[len(tokens)-1]
 }
 
-// GetVersion of VM for SD-based platforms
-func (v *VMSD) GetVersion() string {
-	return v.Version
-}
-
-// GetURL of VM for SD-based platforms
-func (v *VMSD) GetURL() string {
-	return v.URL
-}
-
-// Dir of VM for SD-based platforms
-func (*VMSD) Dir() string {
-	sdRepo := filepath.Join(vboxDir, "sd")
-	return sdRepo
-}
-
-// Name of VM for SD-based platforms
-func (v *VMSD) Name() string {
-	tokens := strings.Split(v.URL, "/")
-	return tokens[len(tokens)-1]
+// S3RepositoryVM is a configuration entry for all VMs
+type VMRepo struct {
+	Vms struct {
+		Vm struct {
```

Comment:
__[golint]__
* [golint] struct field Vm should be VM

<!-- Comment by [SideCI](https://sideci.com) -->